tests/std/tests/P0528R3_cmpxchg_pad/test.cpp

```diff
@@ -0,0 +1,137 @@
+#include <atomic>
+#include <cassert>
+
+struct X1 {
+    char x : 6;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+struct X2 {
+    short x : 9;
+
+    void set(char v) {
+        x = v;
+    }
+};
+
+#pragma pack(push, 1)
+struct X3 {
+    char x : 4;
+    char : 2;
+    char y : 1;
+    short z;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+#pragma pack(pop)
+
+#pragma warning(push)
+#pragma warning(disable : 4324) // '%s': structure was padded due to alignment specifier
+struct alignas(4) X4 {
+    char x;
+
+    void set(char v) {
+        x = v;
+    }
+};
+#pragma warning(pop)
+
+struct X8 {
+    char x;
+    long y;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+    }
+};
+
+struct X16 {
+    long x;
+    char y;
+    long long z;
+
+    void set(char v) {
+        x = v;
+        y = 0;
+        z = 0;
+    }
+};
+
+struct X20 {
+    long x;
+    long y[3];
+    char z;
+
+    void set(char v) {
+        x = v;
+        memset(&y, 0, sizeof(y));

```

Comment:
Need to include `<string.h>` for `memset`, or need to include `<cstring>` and say `std::memset`.